node/src/test/scala/coop/rchain/node/RhoExprToParSpec.scala

```diff
@@ -3,38 +3,122 @@ package coop.rchain.node
 import cats.syntax.all._
 import com.google.protobuf.ByteString
 import coop.rchain.models.syntax._
+import coop.rchain.models.testImplicits._
 import coop.rchain.node.api.WebApi._
-import org.scalatest.wordspec.AnyWordSpec
+import org.scalacheck.Arbitrary
+import org.scalacheck.Arbitrary._
 import org.scalatest.matchers.should.Matchers
+import org.scalatest.propspec.AnyPropSpec
+import org.scalatestplus.scalacheck.ScalaCheckDrivenPropertyChecks
+
+class RhoExprToParSpec extends AnyPropSpec with ScalaCheckDrivenPropertyChecks with Matchers {
 
-class RhoExprToParSpec extends AnyWordSpec with Matchers {
   private def roundTripTest(expr: RhoExpr): Option[RhoExpr] =
     exprFromParProto(rhoExprToParProto(expr))
 
-  "WebAPI RhoExpr <-> Par conversion" should {
-    "work correct with simple types" in {
-      roundTripTest(ExprBool(true)).collect { case ExprBool(data)            => data } shouldBe true.some
-      roundTripTest(ExprInt(42)).collect { case ExprInt(data)                => data } shouldBe 42.some
-      roundTripTest(ExprString("abc")).collect { case ExprString(data)       => data } shouldBe "abc".some
-      roundTripTest(ExprUri("http://test.com")).collect { case ExprUri(data) => data } shouldBe "http://test.com".some
+  // Simple types
+
+  property("WebAPI ExprBool <-> Par conversion work correct") {
```

Comment:
You don't need to do each case manually. Leave this to ScalaCheck to do. Just define Arbitrary[RhoExpr] and check `RhoExpr <=> RhoExpr`.